app/api/v1/vaults.py

```
@@ -1,14 +1,51 @@
+    owner_id: str = Depends(get_current_user_id),
+    repo: VaultRepository = Depends(get_vault_repo),
+) -> ProvisionVaultResponse:
+    use_case = ProvisionVault(repo)
+    try:
+        result = use_case.execute(
+            owner_id=owner_id,
+            hardware_uuid=payload.hardware_uuid,
+            nickname=payload.nickname,
+        )
+    except HardwareAlreadyProvisioned:
+        raise HTTPException(status_code=409, detail="Hardware already provisioned")
```

Ruff (0.14.11)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)